node/full.go

```diff
@@ -292,16 +269,25 @@ func (n *FullNode) OnStart() error {
 	if err = n.fraudService.Start(n.ctx); err != nil {
 		return fmt.Errorf("error while starting fraud exchange service: %w", err)
 	}
+	n.blockManager.SetFraudProofService(n.fraudService)
 
 	if n.conf.Aggregator {
 		n.Logger.Info("working in aggregator mode", "block time", n.conf.BlockTime)
 		go n.blockManager.AggregationLoop(n.ctx, n.conf.LazyAggregator)
 		go n.headerPublishLoop(n.ctx)
 	}
+	go n.blockManager.ProcessFraudProof(n.fraudService, n.ctx, n.cancel)
 	go n.blockManager.RetrieveLoop(n.ctx)
 	go n.blockManager.SyncLoop(n.ctx, n.cancel)
-	go n.fraudProofPublishLoop(n.ctx)
-
+	go func(ctx context.Context) {
+		select {
+		case <-ctx.Done():
+			if err = n.Stop(); err != nil {
+				n.Logger.Error("errors while stopping node:", "errors", err)
+			}
+		default:
+		}
+	}(n.ctx)
```

Comment:
is the point of the go routine to just catch errors on stop?

if so we should not use a select statement and just block on the `<-ctx.Done()` otherwise this go routine just immediately executes and returns right? 

nit: we are inconsistent in scope handling here right?

we are passing in the ctx but then directly referencing the node. 

so to address both questions I would recommend.

```suggestion
	go func(node *FullNode) {
		<-node.ctx.Done():
		if err = node.Stop(); err != nil {
			node.Logger.Error("errors while stopping node:", "errors", err)
		}
	}(n)
```

Reply:
have to remove this code, if the nodes stop after verifying the fraud, they won't be able to serve any other full nodes that want to sync the fraud proof. what should be the intended behavior is more of a design question than engineering. @Manav-Aggarwal @nashqueue